packages/block-editor/src/components/writing-flow/use-tab-nav.js

```diff
@@ -102,6 +102,7 @@ export default function useTabNav() {
 			const direction = isShift ? 'findPrevious' : 'findNext';
 
 			if ( ! hasMultiSelection() && ! getSelectedBlockClientId() ) {
+				setNavigationMode( true );
```

Comment:
How is this related to this PR? What's the purpose of this change?

Reply:
It is needed to make e2e tests pass. We enable navigation mode when the user tabs into the writing flow focus capture, but with tab index set to `-1` and when clearing block selection, focus will no longer move to the interface content element, but rather to the editor styles wrapper. When tabbing, it will not go through to focus capture to enable navigation mode, so we have to enable navigation mode when the user tabs in the content and there is no block selection.

Reply:
I don't think I follow here, why do we need to auto-enable navigation mode implicitly (not hitting "Escape"). 

If there's no block selected and we tab into the content, can't we just select the first one?

Reply:
We could, but that's not how the behaviour was. I want to preserve the current behaviour. This was discussed once with the some people from the accessibility team that if no block is selected, and you tab into the content, the editor should be in navigation mode to select a block. Supposedly it wouldn't make sense to decide for the user and pick the first one from which they then have to press Esc to pick another one. But all of this is a bit more opinionated, so for this PR I just want to make the tests pass and not change the current behaviour. :)

Reply:
👍 I didn't know this was the current behavior, that's fine by me.